cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -653,20 +744,36 @@ class two_table_comparator {
    * `j` of the left table.
    *
    * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
+   * @tparam PhysicalElementComparator A relational comparator functor that compares individual
+   * values rather than logical elements, defaults to `NaN` aware relational comparator that
+   * evaluates `NaN` as greater than all other values.
    * @param nullate Indicates if either input column contains nulls.
-   * @return A binary callable object
+   * @return A binary callable object.
    */
-  template <typename Nullate>
-  less_comparator<strong_index_comparator_adapter<device_row_comparator<Nullate>>>
-  device_comparator(Nullate nullate = {}) const
+  template <typename Nullate,
+            typename PhysicalElementComparator = sorting_physical_element_comparator>
+  auto device_less_comparator(Nullate nullate = {}) const noexcept
```

Comment:
I actually liked the idea of dropping the "device" and "comparator" parts from this naming since it isn't really descriptive.

I'd advocate for mimicking the [`std::` functional objects](https://en.cppreference.com/w/cpp/utility/functional) :
- `two_table_comparator::device_less_comparator -> two_table_comparator::less`
- `two_table_comparator::device_less_equivalent_comparator -> two_table_comparator::less_equivalent`

Reply:
What would you like to do for the equality `two_table_comparator`? Right now it is still `device_comparator` rather than `equality` or `equal`

Reply:
From the STL analogy, it would be named `equal_to`. I am happy with that name.